hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseCommitActionExecutor.java

```diff
@@ -226,6 +226,7 @@ protected void commit(HoodieWriteMetadata<O> result, List<HoodieWriteStat> write
     try {
       HoodieActiveTimeline activeTimeline = table.getActiveTimeline();
       HoodieCommitMetadata metadata = result.getCommitMetadata().get();
+      metadata = appendMetadataForMissingFiles(metadata);
```

Comment:
Can we fix the result along with its instantiation instead?

Reply:
Based on offline discussion, this could be a bigger and complex change. We will go ahead with the current change.

Reply:
Is this for table version 6 only?

Reply:
yes. 

```
  @Override
  protected HoodieCommitMetadata appendMetadataForMissingFiles(HoodieCommitMetadata commitMetadata) throws IOException {
    if (table.getMetaClient().getTableConfig().getTableVersion().greaterThanOrEquals(HoodieTableVersion.EIGHT)) {
      // reconciliation not required for table version 8 and above.
      return commitMetadata;
    } else {
      return CommitMetadataUtils.reconcileMetadataForMissingFiles(table, getCommitActionType(), instantTime, commitMetadata,
          config, context, storageConf, this.getClass().getSimpleName());
    }
  }
```